MaterialDesignThemes.Wpf/Themes/MaterialDesignTheme.NumericUpDown.xaml

```diff
@@ -0,0 +1,58 @@
+﻿<ResourceDictionary xmlns="http://schemas.microsoft.com/winfx/2006/xaml/presentation"
+                    xmlns:po="http://schemas.microsoft.com/winfx/2006/xaml/presentation/options"
+                    xmlns:x="http://schemas.microsoft.com/winfx/2006/xaml"
+                    xmlns:converters="clr-namespace:MaterialDesignThemes.Wpf.Converters"
+                    xmlns:local="clr-namespace:MaterialDesignThemes.Wpf"
+                    xmlns:system="clr-namespace:System;assembly=mscorlib"
+                    xmlns:transitions="clr-namespace:MaterialDesignThemes.Wpf.Transitions"
+                    xmlns:wpf="clr-namespace:MaterialDesignThemes.Wpf">
+  <ResourceDictionary.MergedDictionaries>
+    <!--
+      we only bring in the dictionaries for controls which were "invented"
+      as part of Material Design in XAML Toolkit. Other themes, for existing controls
+      must be selected manually by the user, so that they override default styles.
+      The easiest way to do this is include:
+      
+      MaterialDesignTheme.Defaults.xaml
+      in your App.xaml
+    -->
+  </ResourceDictionary.MergedDictionaries>
+  <!-- set up default styles for our custom Material Design in XAML Toolkit controls -->
+  <Style TargetType="{x:Type wpf:NumericUpDown}">
+    <Setter Property="MinWidth" Value="100"/>
+    <Setter Property="Padding" Value="4"/>
+    <Setter Property="Template">
+      <Setter.Value>
+        <ControlTemplate TargetType="{x:Type wpf:NumericUpDown}">
+          <Border Background="{TemplateBinding Background}"
+                  BorderBrush="{TemplateBinding BorderBrush}"
+                  BorderThickness="{TemplateBinding BorderThickness}"
+                  Padding="{TemplateBinding Padding}"
+                  CornerRadius="10"
+                  ClipToBounds="True">
+            <DockPanel>
+
+              <RepeatButton x:Name="PART_IncreaseButton"
+                            Content="{local:PackIcon Kind=Plus}"
+                            DockPanel.Dock="Right"
+                            Padding="8,4"
+                            Command="{TemplateBinding IncreaseCommand}"/>
+              <RepeatButton x:Name="PART_DecreaseButton"
+                            Content="{local:PackIcon Kind=Minus}"
+                            DockPanel.Dock="Left"
+                            Padding="8,4"
+                            Command="{TemplateBinding DecreaseCommand}"/>
+              
+              <TextBox x:Name="PART_TextBoxField"
```

Comment:
We probably also want to forward the TextBox hint properties. These are attached properties that should be forwarded into the TextBox so that consumers can change them.
https://github.com/MaterialDesignInXAML/MaterialDesignInXamlToolkit/blob/master/MainDemo.Wpf/Fields.xaml#L107-L118

Reply:
I think, similar to some of the other controls that "host" a `TextBox`, we probably want to strip that particular `TextBox` of all the fancy MDIX stuff and implement it directly in the template for the `NumericUpDown`.

In other words, I assume this should eventually include all the prefix-/suffix-texts, leading-/trailing-icons, `SmartHint`, etc. If you look at some of the other controls (`ComboBox` for example), you will see that those things are handled in the "host" template, and not leveraging a "styled" `TextBox`.

To elaborate on the above. Today we cannot easily leverage a "styled" `Textbox` with the way our triggers are working today I believe, and therefore we end up having to duplicate a lot of the stuff (you'll see that in the other control styles). It may be a future refactoring task to see if we can somehow leverage a "styled" `TextBox` in the controls that nests one of those rather than having to handle all the fancy stuff in the "host" template.

Reply:
Perfect, I will see about getting that change done.

Reply:
Deciding to punt on doing this for now and not having a hint on this control to start.